### DISCUSSION TITLE
Pmetrics on HPC

### DISCUSSION THREAD

**Discussion by doroft** [LAPKB/Pmetrics#147] [General]
We are trying to get Pmetrics up and running on a  university HPC cluster. It runs well in the interactive mode, which is slow due to restricted resources.  This is to help people who have  very complex models with differential equations. In the BATCH run HPC method (slurm /conda wrapper with R/Pmetrics 2.02) the R coding stage works. NPscript executable does not initiate and the following executables do not propagate if the NPscript is run manually. Has anyone come across this and have any ideas/solutions.  Can Pmetrics work in the BATCH HPC method?

**Comment by mhovd** [comment]
We have previous experience with running Pmetrics on a HPC cluster.
The simplest fix to your issue may be to include `./NPscript` in your run script.
Prior to Pmetrics 2.0 we optimized the previous version (1.5.7) for running on a HPC, see [this document](https://www.sigma2.no/sites/default/files/imce/AUS/Pmetrics_AUS_report.pdf).

Do you have any more information, such as the code you used so far?
Since you asked the question there has also been some development on a new engine for Pmetrics, written in Rust. This does not have any limitations with regards to memory, number of subjects, etc. Perhaps this could be of interest?
